### PR TITLE
Drop support for GHC 7.10.3

### PR DESCRIPTION
**Pull request by Gabriel439**
The immediate motivation for doing this is so that we don't need to
introduce yet another GHC 7.10.3 workaround for
https://github.com/dhall-lang/dhall-haskell/pull/1640

More generally, the main reason we were still supporting GHC 7.10.3
was for Eta compatibility, but Eta appears to no longer be actively
developed.  Given that Eta seems unlikely to support GHC 8 and we
don't plan on supporting GHC 7.10.3 in perpetuity it's probably
time to drop support.

We could probably drop support for additional GHC versions, but this
is just the smallest change necessary to unblock #1640.

### CODE REVIEW

File: dhall/src/Dhall/Tags.hs [hunk 1]
```diff
@@ -37,9 +37,9 @@ import qualified System.Directory as SD
 -}
 
 data LineColumn = LC 
-    { lcLine :: Int
+    { _lcLine :: Int
       -- ^ line number, starting from 1, where to find the tag
-    , lcColumn :: Int
+    , _lcColumn :: Int
```
[the remark this answers is not in the document]
Comment (Gabriel439) [reply dhall/src/Dhall/Tags.hs]: These unused field labels generated a warning specific to GHC 8.0 (which seems to not be present in GHC 8.4), so adding an underscore to unused identifiers is one way to silence the warning

**Comment by jneira** [comment]
I've been waiting to see a specific change that needed or was easier to implement droping support for ghc-7.10.3 to advise to do it so i totally agree on this (i missed #1640 was blocked on that).
Moreover, eta has the option of apply automatically patches to packages to make them buildables. Otoh it already has support for some ghc features of versions greater than 7.10.3.

@sjakobi well, i really like dhall and that was the reason to try keep it buildable with eta and use it as the main config format for etlas so my intent is to keep contributing. You are doing a great job so the project is going forward at a good pace. Otoh, i am lately more involved in the haskell ide ecosystem but i continue keeping an eye on dhall :wink:

**Comment by jneira** [comment]
Thinking twice there are one caveat: i used eta to make the [java language binding](https://github.com/eta-lang/dhall-eta). Keeping it in sync could take more effort cause we should patch dhall to make it buildable with eta via eta-hackage.
However i am not currently maintaining it and i am not sure if it is used at all (maybe @patrickmn?)

**Comment by patrickmn** [comment]
@jneira I do use the dhall libraries with Eta, but agree with @Gabriel439 that it's not sustainable to directly support 7.10.3 forever. Somebody else maintaining Eta patches for later versions seems good! I'm too time-constrained right now, but I might take that on Later(TM).

**Comment by Gabriel439** [comment]
As a side note, in the course of this pull request I enabled Appveyor support for canceling old builds when new revisions come in for pull requests, which will improve Appveyor efficiency

@sjakobi: My plan is to change the bound on `base` in #1640

**Pull request merged**